synapse/storage/events.py

```diff
@@ -1344,6 +1360,186 @@ def _store_redaction(self, txn, event):
             (event.event_id, event.redacts)
         )
 
+    def _compute_chunk_id_txn(self, txn, room_id, event_id, prev_event_ids):
+        """Computes the chunk ID and topological ordering for an event.
+
+        Also handles updating chunk_graph table.
+
+        Args:
+            txn,
+            room_id (str)
+            event_id (str)
+            prev_event_ids (list[str])
+
+        Returns:
+            tuple[int, int]: Returns the chunk_id, topological_ordering for
+            the event
+        """
+
+        # We calculate the chunk for an event using the following rules:
+        #
+        # 1. If all prev events have the same chunk ID then use that chunk ID
+        # 2. If we have none of the prev events but do have events pointing to
+        #    it, then we use their chunk ID if:
+        #     - They’re all in the same chunk, and
+        #     - All their prev events match the events being inserted
+        # 3. Otherwise, create a new chunk and use that
+
+        # Set of chunks that the event refers to. Includes None if there were
+        # prev events that we don't have (or don't have a chunk for)
+        prev_chunk_ids = set()
+
+        for eid in prev_event_ids:
+            chunk_id = self._simple_select_one_onecol_txn(
+                txn,
+                table="events",
+                keyvalues={"event_id": eid},
+                retcol="chunk_id",
+                allow_none=True,
+            )
+
+            prev_chunk_ids.add(chunk_id)
+
+        forward_events = self._simple_select_onecol_txn(
+            txn,
+            table="event_edges",
+            keyvalues={
+                "prev_event_id": event_id,
+                "is_state": False,
```

Comment:
I guess there isn't much of a reason to leave the `prev_state` out of this, since its simply an extra constraint. Though we no longer add `prev_state` stuff and in general I like that we are consistent with always using `is_state: False`. (It would in fact be nice to delete that entire column and all entries where it is true)

Reply:
hrm. I guess I misunderstood what `is_state` meant. I assumed it meant that one or other of the events was a state event. But actually it means the edge is a prev_state edge rather than a prev_event edge?

Reply:
Indeed